doc/west/repo-tool.rst

```diff
@@ -0,0 +1,270 @@
+   multi-repo work, not replace it. For tasks that aren't multi-repo-related,
+   use plain Git commands.
+
+   This page explains what the West multi-repo commands do behind the scenes.
+
+Manifests
+=========
+
+A *manifest* is a `YAML <http://yaml.org/>`_ file that gives the URL and
+revision for each repository in the multi-repo (each *project*), possibly along
+with other information. Manifests are normally stored in a separate Git
+repository.
+
+Running ``west init`` to initialize a West installation will clone the selected
+manifest repository. Running ``west init`` without specifying a manifest
+repository will use `a default manifest repository
+<https://github.com/zephyrproject-rtos/manifest>`_ for Zephyr. A different
+manifest repository can be selected with ``west init --manifest-url URL``.
+
+The format of the West manifest is described by a `pykwalify
+<https://pypi.org/project/pykwalify/>`_ schema, found `here
+<https://github.com/zephyrproject-rtos/west/blob/master/src/west/manifest-schema.yml>`_.
+
+The ``manifest-rev`` branch
+***************************
+
+West creates a branch named ``manifest-rev`` in each project, pointing to the
+commit at the project's revision, as specified in the manifest. The
+``manifest-rev`` branch is updated whenever project data is fetched (the
+`command overview`_ below explains which commands fetch project data).
+
+All branches created through West track the ``manifest-rev`` branch. Several
+multi-repo commands also use ``manifest-rev`` as a reference for the upstream
+revision (as of the most recent fetch).
+
+.. note::
+
+   ``manifest-rev`` is a normal Git branch, and is only treated specially by
+   name. If you delete it or otherwise modify it, it will be
+   recreated/reset when upstream data is next fetched by ``west``, as if
+   through ``git reset --hard`` (though ``git update-ref`` is used internally).
+
+   Since ``manifest-rev`` represents the upstream revision as of the most
+   recent fetch, it is normally a bad idea to modify it.
+
+   ``manifest-rev`` was added to allow branches to track SHA revisions, and to
+   give a consistent reference for the upstream revision regardless of how the
+   manifest changes over time.
+
+Command overview
+================
+
+This section gives a quick overview of the multi-repo commands, split up by
+functionality. All commands loosely mimic the corresponding Git command, but in
+a multi-repo context (``west fetch`` fetches upstream data without updating
+local work branches, etc.)
+
+Passing no projects to commands that accept a list of projects usually means to
+run the command for all projects listed in the manifest.
+
+.. note::
+
+   For the most up-to-date documentation, see the command help texts (e.g.
+   ``west fetch --help``). Only the most important flags are mentioned here.
+
+Cloning and updating projects
+*****************************
+
+After running ``west init`` to initialize West (e.g., with the default Zephyr
```

Comment:
Do we need the "(e.g....)" ?

Reply:
Used to say "after running ``west init`` to initialize Zephyr". That part was added as a compromise.

I could go back to the original version too, though I think it would have confused me.